src/utils/matches.ts

```diff
@@ -1,7 +1,9 @@
+import path from "path"
 import micromatch from "micromatch"
 
 export default function matches(paths: string[], patterns: string[]): string[] {
+	const isBasename = patterns.every((pattern) => path.basename(pattern) === pattern)
```

Comment:
nit: Could we add a short comment about this workaround so it has a bit of context?

Reply:
Also, IIUC this might have unexpected behavior if some patterns are fully qualified and some are not -- am I reading this right?

Reply:
commented! and yeah hmmmm. I definitely chose the easy way out here. Most of the time everything only has one pattern. Maybe I should check and throw an error / warning if they're mixed?

Reply:
Yeah, that sounds good!

Reply:
Done. Looking a little clumsy, but I think much better that than having it silently / confusingly fail.